validation/src/test/scala/hmda/validation/rules/lar/quality/Q035Spec.scala

```diff
@@ -0,0 +1,39 @@
+package hmda.validation.rules.lar.quality
+
+import hmda.model.fi.lar.LoanApplicationRegister
+import hmda.validation.rules.EditCheck
+import hmda.validation.rules.lar.{ BadValueUtils, LarEditCheckSpec }
+import org.scalacheck.Gen
+
+class Q035Spec extends LarEditCheckSpec with BadValueUtils {
+
+  property("All lars with purchaser types not equal to 1 or 3 must pass") {
+    forAll(larGen) { lar =>
+      whenever(lar.purchaserType != 1 && lar.purchaserType != 3) {
+        lar.mustPass
+      }
+    }
+  }
+
+  val purchaserGen: Gen[Int] = Gen.oneOf(1, 3)
+
+  property("Lars with purchaser type equal to 1 or 3 must have loan type equal to 1") {
+    forAll(larGen, purchaserGen) { (lar, x) =>
+      val newLoan = lar.loan.copy(loanType = 1)
+      val newLar = lar.copy(loan = newLoan, purchaserType = x)
+      newLar.mustPass
+    }
+  }
+
+  val irrelevantLoanType: Gen[Int] = intOtherThan(1)
```

Comment:
Small complaint: I think "irrelevant" doesn't work here because Loan Type is not a precondition to the check. `validLoanType` would work.